rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/OneScanActivity.java

```diff
@@ -61,28 +57,65 @@ private void doScan(String reader) {
 
         oneScanHelper.send(request, (resultCode, bundle) -> {
             if (resultCode == Activity.RESULT_OK) {
-                OneScanHelper.ScanResponse response = new OneScanHelper.ScanResponse(bundle);
-                resultView.setText(String.format("\tStatus: %s\n\n\tBarcode Text: %s" +
-                                "\n\n\tProduct ID: %s\n\n\tSerial No: %s\n\n\tAdditional ID: %s\n\n" +
-                                "\tLot: %s\n\n\tExpiration Date: %s\n\n\tSensor triggered: %s",
-                        response.status,
-                        response.barcodeText,
-                        response.productId,
-                        response.serialNumber,
-                        response.additionalIdentifier,
-                        response.lot,
-                        response.expirationDate,
-                        response.sensorTriggered ? "yes" : "no"));
-
-                setResultAndFinishDelayed(response, BARCODE_DETAILS_DISPLAY_TIMEOUT);
+                performPostScanActions(bundle);
             }
         });
     }
 
+    private void performPostScanActions(Bundle bundle) {
+        response = new OneScanHelper.ScanResponse(bundle);
+        try {
+            displayStatusMessageAndSymbol(response.sensorTriggered, response.expirationDate);
+            populateBarcodeDetails();
+        } catch (ParseException e) {
+            Timber.e(e);
+        }
+    }
+
+    private void populateBarcodeDetails() {
+        setBarcodeResult(R.id.barcode_product_id, response.productId);
+        setBarcodeResult(R.id.barcode_serial_no, response.serialNumber);
+        setBarcodeResult(R.id.barcode_additional_id, response.additionalIdentifier);
+        setBarcodeResult(R.id.barcode_lot_no, response.lot);
+        setBarcodeResult(R.id.barcode_expiration_date, response.expirationDate);
+        setBarcodeResult(R.id.barcode_is_sensor_triggered, response.sensorTriggered ? "Yes" : "No");
```

Comment:
do we need to load this text somehow so it is translated properly?

Reply:
Done